pkg/controller/siddhiprocess/deploymanager/application.go

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright (c) 2019 WSO2 Inc. (http:www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http:www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package deploymanager
+
+import (
+	"strconv"
+
+	siddhiv1alpha2 "github.com/siddhi-io/siddhi-operator/pkg/apis/siddhi/v1alpha2"
+	artifact "github.com/siddhi-io/siddhi-operator/pkg/controller/siddhiprocess/artifact"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	"gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"path/filepath"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// Application contains details about the siddhi app which need in the K8s deployment
+type Application struct {
+	Name               string                 `json:"appName"`
+	ContainerPorts     []corev1.ContainerPort `json:"containerPorts"`
+	Apps               map[string]string      `json:"apps"`
+	ServiceEnabled     bool                   `json:"serviceEnabled"`
+	PersistenceEnabled bool                   `json:"persistenceEnabled"`
+	Replicas           int32                  `json:"replicas"`
+}
+
+// Image holds the docker image details of the deployment
+type Image struct {
+	Name    string
+	Home    string
+	Secret  string
+	Profile string
+}
+
+// DeployManager creates and manage the deployment
+type DeployManager struct {
+	Application   Application
+	KubeClient    artifact.KubeClient
+	Labels        map[string]string
+	Image         Image
+	SiddhiProcess *siddhiv1alpha2.SiddhiProcess
+}
+
+// SPConfig contains the state persistence configs
+type SPConfig struct {
+	Location string `yaml:"location"`
+}
+
+// StatePersistence contains the StatePersistence config block
+type StatePersistence struct {
+	SPConfig SPConfig `yaml:"config"`
+}
+
+// Config contains the siddhi config block
+type Config struct {
+	StatePersistence StatePersistence `yaml:"state.persistence"`
+}
+
+// CreateLabels creates labels for each deployment
+func (d *DeployManager) CreateLabels() {
+	labels := map[string]string{
```

Comment:
Do we need this method `CreateLabels`? can't we use a function to generate the label without the DeployManager receiver? We only change the `siddhi.io/instance` key which can be an argument to that function.

Reply:
But here we also assign the labels generated labels to the deployment manager instance. So, I think it would be better to have it with `DeployManager`.